source/extensions/omni.isaac.lab/omni/isaac/lab/envs/mdp/observations.py

```diff
@@ -279,65 +283,98 @@ def image(
+
+    Args:
+        sensor_cfg: The sensor configuration to poll. Defaults to SceneEntityCfg("tiled_camera").
+        data_type: The sensor data type. Defaults to "rgb".
+        convert_perspective_to_orthogonal: Whether to orthogonalize perspective depth images.
+            This is used only when the data type is "distance_to_camera". Defaults to False.
+        model_zoo_cfg: A user-defined dictionary that maps different model names to their respective configurations.
+            Defaults to None. If None, the default model zoo configurations are used.
+        model_name: The name of the model to use for inference. Defaults to "ResNet18".
+        model_device: The device to store and infer the model on. This is useful when offloading the computation
+            from the environment simulation device. Defaults to the environment device.
+
+    Returns:
+        The extracted features tensor. Shape is (num_envs, feature_dim).
+
+    Raises:
+        ValueError: When the model name is not found in the provided model zoo configuration.
+        ValueError: When the model name is not found in the default model zoo configuration.
     """
 
     def __init__(self, cfg: ObservationTermCfg, env: ManagerBasedEnv):
+        # initialize the base class
         super().__init__(cfg, env)
-        from torchvision import models
-        from transformers import AutoModel
 
-        def create_theia_model(model_name):
-            return {
-                "model": (
-                    lambda: AutoModel.from_pretrained(f"theaiinstitute/{model_name}", trust_remote_code=True)
-                    .eval()
-                    .to("cuda:0")
-                ),
-                "preprocess": lambda img: (img - torch.amin(img, dim=(1, 2), keepdim=True)) / (
-                    torch.amax(img, dim=(1, 2), keepdim=True) - torch.amin(img, dim=(1, 2), keepdim=True)
-                ),
-                "inference": lambda model, images: model.forward_feature(
-                    images, do_rescale=False, interpolate_pos_encoding=True
-                ),
-            }
-
-        def create_resnet_model(resnet_name):
-            return {
-                "model": lambda: getattr(models, resnet_name)(pretrained=True).eval().to("cuda:0"),
-                "preprocess": lambda img: (
-                    img.permute(0, 3, 1, 2)  # Convert [batch, height, width, 3] -> [batch, 3, height, width]
-                    - torch.tensor([0.485, 0.456, 0.406], device=img.device).view(1, 3, 1, 1)
-                ) / torch.tensor([0.229, 0.224, 0.225], device=img.device).view(1, 3, 1, 1),
-                "inference": lambda model, images: model(images),
-            }
-
-        # List of Theia models
-        theia_models = [
+        # extract parameters from the configuration
+        self.model_zoo_cfg: dict = cfg.params.get("model_zoo_cfg")  # type: ignore
+        self.model_name: str = cfg.params.get("model_name", "ResNet18")  # type: ignore
```

Comment:
Case of this doesn't match the case of resnet list (ResNet18 vs. resnet)

Reply:
You're looking at an outdated version :)